addons/hr_recruitment/models/hr_applicant.py

```diff
@@ -366,6 +366,12 @@ def write(self, vals):
                 vals['kanban_state'] = 'normal'
             for applicant in self:
                 vals['last_stage_id'] = applicant.stage_id.id
+                new_stage = self.env['hr.recruitment.stage'].browse(vals['stage_id'])
+                if new_stage.hired_stage and not applicant.stage_id.hired_stage:
+                    if applicant.job_id.no_of_recruitment > 0:
+                        applicant.job_id.no_of_recruitment -= 1
+                elif not new_stage.hired_stage and applicant.stage_id.hired_stage:
+                    applicant.job_id.no_of_recruitment += 1
                 res = super().write(vals)
         else:
             res = super().write(vals)
```

Comment:
unrelated, but it could just be
```suggestion
        res = super().write(vals)
```